src/App.js

```diff
@@ -1,16 +1,31 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
 
 const App = () => {
+
+  const [likes, numLikes] = useState(0);
+  const updateLikes = (clicked, numLike) => {
```

Comment:
So, there are a couple things that I want to discuss here, because I think that there is some confusion when it comes to lifting up state. Here what you are doing is making a `likes` state and getting the accompanied `numLikes`. Then inside of `updateLikes` you have parameters `clicked` and `numLike` which will be passed in from `chatEntry.js` where they manage each entry's `like` state.  But if you recall, each message has their own `liked` key/value pair. This what we actually want to change and we will do that by establishing `state` with the messages data inside of the `App.js` file and creating an `updateLikes` function that will be passed down to each `chatEntry` who will then pass their `id` so that the corresponding `like` key/value pair can be updated. Finally, we can create a `calculateTotal` function that will total the likes each render. The refactor will take place over multiple files and would look something like this:

```javascript
// APP.JS CHANGES 

const [entries, setEntries] = useState(chatMessages)

  const updateLikes = (id) => {
    const newEntries = entries.map(entry => {
      if (entry.id === id) return {...entry, liked: !entry.liked}
      return entry
    })
    setEntries(newEntries)
    }

    const calculateHearts = () => {
      let total = 0;
      for (let entry of entries) {
        if (entry.liked === true) {
          total += 1;
        }
      }
      return total;
    };

 <p>{calculateHearts()} ❤️s</p> // This line runs each render, since state is in App.js whenever it changes it will cause the `App.js` and all its child components to re-render.

// CHATLOG.JS CHANGES 

const ChatLog = ({entries,updateLikes}) => {

    return (
    <div className="chat-log">
        {entries.map((entry) => (
            <ChatEntry 
            sender={entry.sender}
            body = {entry.body}
            timeStamp = {entry.timeStamp} 
            updateLikes ={updateLikes}
            liked={entry.liked} // We pass this property in because now it is being changed in our entries state.
            id={entry.id} // id needs to be passed so we can use it to identify which entry is causing this change
            />
        ))}
    </div>  
    );         
};

// CHATENTRY.JS CHANGES

 <button className="like" onClick={()=> updateLikes(id)}>
          {liked ? '❤️': '🤍' }v // when the user clicks the heart the id will be used to update the entries liked key/value with will then cause it to render based on if the liked prop is now true or false

```

This is a lot of changes across numerous files, so please feel free to reach out to me if you need further explanation.

Reply:
i definitely have some question will schedule a meeting